### PR TITLE
🥋 Add proposal details in vote on proposal modal

### LINKED ISSUE #1549: Vote on proposal modal
Add the ["vote on proposal"](https://www.figma.com/file/pgNl83wgAub186krVkdiuo/Proposals?node-id=338%3A12916) modal.

The modal:

*   [x] Should validate required stake for transaction
*   [x] Vote Step
    *   [x] Should have 3 buttons: Approve, Reject, Abstain
    *   [x] Only one button can be selected at a time
    *   [x] Selecting Reject should show also the "Slash Proposal" checkbox as well (default "no")
*   [x] Proposal general parameters on the left side (title, rationale)
*   [x] Rationale input
    *   [x] check & validate maximum rationale length
*   [x] Sign and send transaction step

Authorize step:

### PR DESCRIPTION
**Pull request by krupka4soft**
Closes #1549

### CODE REVIEW

File: packages/ui/src/proposals/modals/VoteForProposal/components/ProposalPreview/ProposalPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    if (Component) {
+      return <Component label={detail.label} value={detail.value} key={index} />
+    }
+
+    return null;
+  }, [])
+
+  if (!proposalDetails) { return null; }
+
+  const detailsRenderStructure = getDetailsRenderStructure(proposalDetails)
```
Comment (krupka4soft) [inline packages/ui/src/proposals/modals/VoteForProposal/components/ProposalPreview/ProposalPreview.tsx:49]: `useMemo`

File: packages/ui/src/proposals/modals/VoteForProposal/components/ProposalPreview/renderers/Address.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+interface Props {
+  label: string
+  value: string
+}
+
+export const Address = ({
+  label,
+  value,
+}: Props) => {
+  return (
```
Comment (krupka4soft) [inline packages/ui/src/proposals/modals/VoteForProposal/components/ProposalPreview/renderers/Address.tsx:18]: skip `return`

**Review by krupka4soft — COMMENTED** [review]
self CR done

**Comment by jodator** [comment]
@krupka4soft The other PR that touched the proposal details was merged. Please check the conflicts as well.

ps.: Remember to link the issue in the PR message (it is blank now).

**Comment by jodator** [comment]
@p-sad I added you as reviewer, so you can check the proposed changes to the way how we render proposal props.
Also, please wait after @krupka4soft will update the remaining changes.

**Review by jodator — APPROVED** [review]
Should be OK 👍 as we were pair-programming most of the time.

**Review by p-sad — APPROVED** [review]

**Pull request merged**